src/Blake.BuildTools/Utils/PluginLoader.cs

```diff
@@ -169,10 +169,46 @@ private static bool IsNuGetPluginValid(NuGetPluginInfo plugin, ILogger? logger)
             }
 
             // For NuGet packages, the file version should match the package version
-            // Some packages may have different versioning schemes, so we'll be lenient
-            if (fileVersion.StartsWith(plugin.Version))
+            // Compare versions component-wise for leniency, but avoid false positives
+            if (Version.TryParse(plugin.Version, out var expectedVersion) && Version.TryParse(fileVersion, out var actualVersion))
```

Comment:
This is great but I think it should be moved to an extension method or a utility class.